src/fastoad/notebooks/02_explanation/04_OpenMDAO.ipynb

```diff
@@ -0,0 +1,892 @@
+{
```

Comment:
<p>I'd rather begin by explaining the purpose of groups: assembling components (for convenience), hiding variables (by not promoting them), declaring connections, or defining sub-loops/sub-solvers (anything else?). Then we declare that groups are used here for convenience (and for tutorial purposes) and the explanation can go on...</p><p> </p><blockquote>It can sometime happen<strong><s>s</s></strong> that when setting up a group, ...</blockquote><p> </p><blockquote>This is gonna be the case...</blockquote><p>--&gt; "This will be the case..."</p><p> </p><blockquote>When this occurs, long story short, OpenMDAO detects that there is a loop and it solves it <strong>iterativey</strong> when a solver is added to said group.</blockquote><p>Besides the typo, I'd rather focus on the need for setting a solver to solve the loop (as a beginner, I overlooked that sometimes), and then tell that OpenMDAO will automatically "detect" the variables that are looped on.</p><p> </p><blockquote>Remark: </blockquote><blockquote><br></blockquote><blockquote>If we had a component that had options, we would have need to fill them when instantiating said component in the <code>add_subsytem</code> function as if they were the inputs of the class constructor. If we take again the example of the ComputeLiftToDragRatio component that we define with an option earlier in the notebook, it means we should have written : </blockquote><blockquote> <code>python </code></blockquote><blockquote><code>        self.add_subsystem( </code></blockquote><blockquote><code>            name="compute_lift_to_drag", subsys=ComputeLiftToDragRatio(low_speed=False), promotes=["*"] </code></blockquote><blockquote><code>        ) </code></blockquote><p>It is not clear where the remark end. Please, quote the whole block (double quote works !):</p><pre class="ql-syntax" spellcheck="false">&gt; #### Remark:
&gt; If we had a component that had options, we would have need to fill them when instantiating said component in the <code>add_subsytem</code> function as if they were the inputs of the class constructor. If we take again the example of the ComputeLiftToDragRatio component that we define with an option earlier in the notebook, it means we should have written :
&gt;
&gt; &gt; <code></code>` python
&gt; &gt;&nbsp;&nbsp;&nbsp;&nbsp;self.add_subsystem(
&gt; &gt;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;name="compute_lift_to_drag", subsys=ComputeLiftToDragRatio(low_speed=False), promotes=["*"]
&gt; &gt;&nbsp;&nbsp;&nbsp;&nbsp;)
&gt; &gt;<code></code>`
</pre><p> </p><blockquote>we can create the <strong>macro-/super-/global</strong>-group&nbsp;</blockquote><p>Please pick juste one of them :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fast-aircraft-design/FAST-OAD/pull/426/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
<blockquote>This <strong>is </strong>will be the case for the MTOW in our example. When this occurs, a solver need<strong>s</strong> to be added to enable the iterative solving of this loop. Without this solver, and even if OpenMDAO can automatically detect the variables that are looped on, the group will simply be executed once, without assurance that the entry of the loop is equal to its exit</blockquote><p>("is" to be removed, "s" to be added)</p><p> </p><p>And even with no mistake in "without assurance that the entry of the loop is equal to its exit", I would suggest:</p><p>"with no guarantee that the input of the loop is equal to its output"</p>

Reply:
<blockquote>we can create the global-group</blockquote><p>Why the hyphen in global<code>-</code>group?</p>

Reply:
<blockquote>The syntax used to add a component inside a group is via the <code>add_subsystem</code> functions in which we must fill : </blockquote><blockquote> - A name for the subsy<code>s</code>tem we are adding. Making it unique can help debugging when OpenMDAO returns errors.</blockquote><p> </p><p>Apart from the typo, maybe I would put some more details here to tell the added subsystem will be reachable as a Python attribute of the parent group, using the chosen name, meaning this name has to be Python-compliant. But the usefulness of having it unique still stands.</p>

Reply:
<p>Thanks for the modifications.</p><p> </p><p>I think now it needs a bit more clarity with "Making it unique for the whole problem can help debugging when errors are returned".</p><p> </p><p>(because we know already it must be unique in the group)</p>

Reply:
<p>Added a bit more detailed in 7a6643b4, by saying that OpenMDAO displays the "path" to the faulty components using the name we provide in the <code>add_subsystem</code> function </p>